dask/dataframe/groupby.py

```diff
@@ -538,17 +538,64 @@ class _GroupBy(object):
 
     obj: DataFrame or Series
         DataFrame or Series to be grouped
-    index: str, list or Series
+    by: str, list or Series
         The key for grouping
+    slice: str, list
+        The slice keys applied to GroupBy result
+    axis: int
+        Not supported
+    level: int, level name, or sequence of such, default None
+        Not supported
+    as_index: bool, default True
+        Not supported, the returned objects always have group
+        labels as the index
+    sort: bool, default True
+        Not supported
+    group_keys: bool, default True
+        Not supported
+    squeeze: bool, default False
+        Not supported
     kwargs: dict
         Other keywords passed to groupby
     """
-    def __init__(self, df, index=None, slice=None, **kwargs):
+    def __init__(self, df, by=None, slice=None, axis=0, level=None,
```

Comment:
this is a little clunky to have to name these args (since they are unsupported). Are these remaining ``kwargs`` actually used anywhere? Whey not simply list the acceptable ones? (and it will show a ``TypeError`` if you pass not named ones)